pkg/operation/care/constraints.go

```diff
@@ -171,10 +171,11 @@ func (c *Constraint) CheckForProblematicWebhooks(ctx context.Context) (gardencor
 	for _, webhookConfig := range validatingWebhookConfigs {
 		for _, w := range webhookConfig.Webhooks {
 			if IsProblematicWebhook(w.FailurePolicy, w.ObjectSelector, w.NamespaceSelector, w.Rules, w.TimeoutSeconds) {
+				msg := buildProblematicWebhookMessage("ValidatingWebhookConfiguration", webhookConfig.Name, w.Name, w.FailurePolicy, w.TimeoutSeconds)
 				return gardencorev1beta1.ConditionFalse,
 					"ProblematicWebhooks",
-					buildProblematicWebhookMessage("ValidatingWebhookConfiguration", webhookConfig.Name, w.Name, w.FailurePolicy, w.TimeoutSeconds),
-					nil
+					msg,
+					gardencorev1beta1helper.NewErrorWithCodes(msg, gardencorev1beta1.ErrorUserWebhook)
```

Comment:
I think you'll also need to make changes here,
https://github.com/gardener/gardener/blob/f7b388e2db1256de4b2254246e7cff7f177946e3/pkg/operation/care/checker.go#L648
`return gardencorev1beta1helper.UpdatedConditionUnknownError(oldCondition, err, gardencorev1beta1helper.ExtractErrorCodes(err)...)`

Reply:
Thank you for the feedback. I have made the changes as per suggestion. But I still need to test the changes.

Reply:
In the [commit](ef5e8575d0a9a41eedf5a0e8a3c1a0732901a8c4), we are returning error codes from `CheckForProblematicWebhooks` function. This what I have got after testing the implementation in local setup.

Case : Problematic webhook present
```
Constraints:
    Codes:
      ERR_USER_WEBHOOK
    Last Transition Time:  2022-04-28T10:15:50Z
    Last Update Time:      2022-04-28T10:15:43Z
    Message:               MutatingWebhookConfiguration "fail-webhook" is problematic: webhook "my-webhook.example.com" with failurePolicy "Fail" and 5s timeout might prevent worker nodes from properly joining the shoot cluster
    Reason:                ProblematicWebhooks
    Status:                False
    Type:                  HibernationPossible
    Codes:
      ERR_USER_WEBHOOK
    Last Transition Time:  2022-04-28T10:15:50Z
    Last Update Time:      2022-04-28T10:15:43Z
    Message:               MutatingWebhookConfiguration "fail-webhook" is problematic: webhook "my-webhook.example.com" with failurePolicy "Fail" and 5s timeout might prevent worker nodes from properly joining the shoot cluster
    Reason:                ProblematicWebhooks
    Status:                False
    Type:                  MaintenancePreconditionsSatisfied
```

Case: Problematic webhook not present
```
Constraints:
    Last Transition Time:  2022-04-28T10:46:56Z
    Last Update Time:      2022-04-28T10:46:14Z
    Message:               All webhooks are properly configured.
    Reason:                NoProblematicWebhooks
    Status:                True
    Type:                  HibernationPossible
    Last Transition Time:  2022-04-28T10:46:56Z
    Last Update Time:      2022-04-28T10:46:14Z
    Message:               All webhooks are properly configured.
    Reason:                NoProblematicWebhooks
    Status:                True
    Type:                  MaintenancePreconditionsSatisfied
```

Is this the right implementation ? @timebertt @grolu

Reply:
Nit: you can mention this in PR description also.